docs/traits.md

```diff
@@ -0,0 +1,152 @@
+# `ArborX::Traits::Access`
+Defined in header [`<ArborX_Traits.hpp>`](https://github.com/arborx/ArborX/blob/master/src/details/ArborX_Traits.hpp)
+
+```C++
+template <typename T, typename Tag, typename Enable=void>
+struct Access;
+```
+The struct template `ArborX::Traits::Access` tells the [`ArborX::BVH` constructor](https://github.com/dalg24/ArborX/blob/docs/docs/bounding_volume_hierarchy.md#arborxbvhdevicetypebvh) how much data to index and how to access that data,
+or [`ArborX::BVH::query()`](https://github.com/dalg24/ArborX/blob/docs/docs/bounding_volume_hierarchy.md#arborxbvhdevicetypequery) what predicates to query for.
+
+ArborX provides the folowing partial specialization for Kokkos views
+```C++
+template <typename View, typename Tag>
+Access<View, Tag, std::enable_if_t<Kokkos::is_view<View>::value>>;
+```
+
+```C++
+struct PrimitivesTag { };
+struct PredicatesTag { };
+```
+Empty class types used to indicate the data category.
+
+## Template parameters
+`T`
+: User data type.  
+`Tag`
+: Tag to distinguish between primitives and predicates.
+
+## Member types
+Member type | Description
+--- | ---
+`memory_space` | A valid Kokkos memory space
+
+## Static member functions
+[`size`](#arborxtraitsaccessttagsize)
+: returns the number of elements  
+[`get`](#arborxtraitsaccessttagget)
+: access the specified element.
+
+## Example
+```C++
+#include <ArborX_LinearBVH.hpp>
+#include <Kokkos_Core.hpp>
+#include <array>
+#include <iostream>
+#include <numeric>
+
+struct PointCloud
+{
+  float* d_x;
+  float* d_y;
+  float* d_z;
+  int N;
+};
+
+namespace ArborX {
+namespace Traits {
+template <>
+struct Access<PointCloud, PrimitivesTag>
+{
+    inline std::size_t size(PointCloud const& cloud)
+    {
+        return cloud.N;
+    }
+      KOKKOS_INLINE_FUNCTION Point get(PointCloud const& cloud, std::size_t i)
```

Comment:
So if later on we decide that `Point` is templated on the dimension, what needs to change?